src/main/scala/org/apache/hadoop/fs/sftpwithseek/SFTPConnectionPool.scala

```diff
@@ -0,0 +1,229 @@
+package org.apache.hadoop.fs.sftpwithseek
+
+import com.jcraft.jsch.ChannelSftp
+import com.jcraft.jsch.JSch
+import com.jcraft.jsch.JSchException
+import com.jcraft.jsch.Session
+import org.apache.commons.logging.Log
+import org.apache.commons.logging.LogFactory
+import org.apache.hadoop.util.StringUtils
+import java.io.IOException
+import java.util
+import java.util.Properties
+
+import org.apache.hadoop.fs.sftpwithseek.SFTPConnectionPool.ConnectionInfo
+
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** Concurrent/Multiple Connections. */
+object SFTPConnectionPool {
+
+  val LOG: Log = LogFactory.getLog(classOf[SFTPFileSystem])
+
+  /**
+   * Class to capture the minimal set of information that distinguish
+   * between different connections.
+   */
+  private[sftpwithseek] class ConnectionInfo private[sftpwithseek] (
+      val hst: String,
+      var port: Int,
+      val usr: String
+  ) {
+
+    private var host = ""
+    private var user = ""
+    this.host = hst
+    this.user = usr
+    def getHost: String            = host
+    def setHost(hst: String): Unit = this.host = hst
+    def getPort: Int               = port
+    def setPort(prt: Int): Unit    = this.port = prt
+    def getUser: String            = user
+    def setUser(usr: String): Unit = this.user = usr
+
+    def canEqual(a: Any) = a.isInstanceOf[ConnectionInfo]
+
+    override def equals(obj: Any): Boolean = {
```

Comment:
If you make this class a case class, you might not need this

Reply:
Same as above.

Reply:
Moved the classes to their original Java implemention in: 25d34a0e31b595893e2f9bd3fe392ba9591ef249